### PR TITLE
Added support for the `haul` react-native packager

### PR DESCRIPTION
**Pull request by ericwooley**
Issue: New Feature

## What I did

I added another cli option to use the haul cli instead of the react-native packager.

## How to test

1. create a react-native new app with a new install of storybook.

2. install haul from hauls app initializer.

3. `cp webpack.haul.js webpack.haul.storybook.js`

4. change the entry from 
```
module.exports = ({ platform }) => ({
  entry: `./index.${platform}.js`,
});
```
to
```
module.exports = ({ platform }) => ({
  entry: `./storybook/index.${platform}.js`,
});
```
5. add ` --haul webpack.haul.storybook.js` to your storybook command
6. run storybook. 

Haul should take over and do the job of the react-native packager.

### CODE REVIEW

**Comment by shilman** [comment]
Nice one, thanks @ericwooley ! PR looks good, I'll test it out later today and let you know if I have questions.

**Review by ndelangen — APPROVED** [review]
PR looks good to me, @shilman this sounds like a breakthrough in regards to the RN + npm linking problem?

File: app/react-native/readme.md [hunk 1]
```diff
@@ -72,6 +72,21 @@ For RN apps:
 
 Once your app is started, changing the selected story in web browser will update the story displayed within your mobile app.
 
+## Using Haul-cli
+
+[Haul](https://github.com/callstack-io/haul) is an alternative to the react-native packager and has several advantages in that it allows you to define your own loaders, and handles symlinks better.
+
+If you want to use haul instead of the react-native packager, modify they storybook npm script too:
```
Comment (orta) [inline app/react-native/readme.md]: Typo: ~too~  to

Comment (orta) [inline app/react-native/readme.md]: And ~they~ the

**Comment by shilman** [comment]
@ndelangen @tmeasday YES! haul does solve our RN simlink problem. tested it out for this PR, and also with lerna bootstrap. great news and thank you @ericwooley for introducing us to `haul`. will be opening a follow-on PR to integrate it into our build process 🚀

**Review by shilman — CHANGES_REQUESTED** [review]
@ericwooley this is super awesome. mind fixing those minor typos that @orta pointed out? then i'll merge.

**Review by shilman — APPROVED** [review]

**Comment by shilman** [comment]
Nevermind. Made the changes myself since I'm so eager to get this merged 😄

**Comment by ericwooley** [comment]
@shilman @ndelangen I'm glad you like it! I was having lerna issues as well which motivated me to investigate!

An interesting thing, after I started using my fork. Stories imported from `node_modules/my_linked_shared_components/stories.js` do not show up in the list of stories. I suspect they are on some kind of ignore list. 

Can anyone point me in the direction of where such code might be? I'll open another PR if I can pinpoint the issue.

File: app/react-native/readme.md (hunk 1 above)
Comment (ericwooley) [inline app/react-native/readme.md]: I'm so bad with spelling. 

I need some kind of ai to just point this kinda stuff out. If only there were some kind of build in tools...

**Comment by slorber** [comment]
great!

Now maybe it's worth to add this to the contribution guide, and maybe create a "test-crna" example that automatically has the proper haul config already setup so that contributors can easily test their changes locally?

**Comment by shilman** [comment]
That's the plan. I'm offline for the next four days but can pick it up when I get back. Or feel free to pick it up and run with it!

File: app/react-native/readme.md (hunk 1 above)
Comment (orta) [inline app/react-native/readme.md]: Yeah, I know this pain, for one of my blogs we use [danger-prose](https://github.com/dbgrandi/danger-prose) to lint for typos.

**Comment by tmeasday** [comment]
Such good news! Do we need to add `haul` as a depenendency?

**Comment by SEAPUNK** [comment]
Doesn't Expo use the react-native packager regardless, though? How does the packaging process go in correlation to running Storybook and Expo w/ `storybook start` and then `react-native-scripts ios`?

**Comment by ericwooley** [comment]
@tmeasday currently I don't think you would, but I am not certain. 
If they user is switching storybook to use haul, they are almost certainly using haul for their app and thus it should be installed. 

There may be some edge case (that I can't think of) where a user wants to use haul for their storybook, and the react-native packager for the actual app, but if thats they case I think they will have to live with adding haul as a dependency manually, instead of requiring everyone install the haul cli, just in case.

@SEAPUNK I'm not super familiar with how expo does it. If they just run `npm run storybook` then I would think it should be fine. If they run a custom command via `./node_modules/.bin/...whatever`, i don't think this would work.

**Comment by ericwooley** [comment]
@tmeasday If there is an argument added to getStorybook, eg `getStorybook --use-haul` it could add that as a dev dependency automatically, and setup the haul webpack file?

**Comment by tmeasday** [comment]
@ericwooley -- I'm not sure what's best, but perhaps we should add it as an `optionalDependency` at the least and show some kind of error if it's not installed? It feels weird/wrong to `import` something that's not in `package.json`.

**Comment by shilman** [comment]
@tmeasday @ericwooley good point about the dependency. i totally missed this in my review. i was too excited about fixing our development/testing environment. 😄

**Pull request merged**